ruby/tests/basic_proto2.rb

```diff
@@ -0,0 +1,1261 @@
+#!/usr/bin/ruby
+
+require 'google/protobuf'
+require 'json'
+require 'test/unit'
+
+# ------------- generated code --------------
+
+module BasicTestProto2
+  file_descriptor = Google::Protobuf::FileDescriptor.new
+  file_descriptor.syntax = :proto2
+
+  pool = Google::Protobuf::DescriptorPool.new
+  pool.build(file_descriptor) do
+    add_message "Foo" do
+      optional :bar, :message, 1, "Bar"
+      repeated :baz, :message, 2, "Baz"
+    end
+
+    add_message "Bar" do
+      optional :msg, :string, 1
+    end
+
+    add_message "Baz" do
+      optional :msg, :string, 1
+    end
+
+    add_message "TestMessage" do
```

Comment:
Instead of creating the message manually, could you use protoc to generate code and import the generated code in test?

Reply:
These tests for proto2 are based on the existing tests for proto3 in `tests/basic.rb`. Are you suggesting to change both to use protoc generated protos (instead of the hardcoded ones)? If the latter, I will prefer to do it in a separate PR.

Reply:
Yes, I mean use generated code (instead of hardcoded) in test. Besides, the test itself should be general to take both prot2 and proto3 message.
Do you plan to have this basic_proto2.rb first, and later refactor it to merge it with test for proto3? I prefer to do the refactoring in this PR.